packages/metrics/src/features/locationStatistics/service.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *
+ * OpenCRVS is also distributed under the terms of the Civil Registration
+ * & Healthcare Disclaimer located at http://opencrvs.org/license.
+ *
+ * Copyright (C) The OpenCRVS Authors. OpenCRVS and the OpenCRVS
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import {
+  fetchLocation,
+  fetchChildLocationsWithTypeByParentId,
+  fetchChildLocationsByParentId,
+  fetchLocationsByType,
+  countUsers
+} from '@metrics/api'
+import { getPopulation, getLocationType } from '@metrics/features/metrics/utils'
+import { IAuthHeader } from '@metrics/features/registration'
+
+interface ILocationStatisticsResponse {
+  registrars: number
+  population?: number
+  offices: number
+}
+
+async function getCRVSOfficeStatistics(
+  location: fhir.Location,
+  authHeader: IAuthHeader
+): Promise<ILocationStatisticsResponse> {
+  const registrars =
+    (await countUsers(
+      { primaryOfficeId: location.id, role: 'LOCAL_REGISTRAR' },
+      authHeader
+    )) || 0
+  return {
+    registrars,
+    offices: 1
+  }
+}
+
+async function getAdminLocationStatistics(
+  location: fhir.Location,
+  populationYear: number,
+  authHeader: IAuthHeader
+): Promise<ILocationStatisticsResponse> {
+  const population = getPopulation(location, populationYear)
+  let offices = 0
+  let registrars = 0
+  const locationOffices = await fetchChildLocationsWithTypeByParentId(
+    location.id as string,
+    'CRVS_OFFICE',
+    authHeader
+  )
+  for (const office of locationOffices) {
+    offices += 1
+    registrars +=
+      (await countUsers(
```

Comment:
Would it be possible to make countUsers an endpoint that would return data like this:

```
[
  {locationId: '...', totalUsers, 123},
  {locationId: '...', totalUsers, 123}
]
```

My concern is that especially with large locations, there might be quite a many offices, which might then mean tens of HTTP requests by this function to user management.

You can do aggregated count by queries in MongoDB like this: https://stackoverflow.com/a/28013417

Reply:
Many thanks, @rikukissa , I will amend the endpoint to fetch location-wise users count to make it be called once